iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/util/RegulatedTerminalControllers.java

```diff
@@ -0,0 +1,223 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.modification.util;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.extensions.Extension;
+import com.powsybl.iidm.network.*;
+import com.powsybl.iidm.network.extensions.RemoteReactivePowerControl;
+import com.powsybl.iidm.network.extensions.SlackTerminal;
+import com.powsybl.iidm.network.extensions.VoltageRegulation;
+
+import java.util.*;
+
+/**
+ * @author Luma Zamarreño {@literal <zamarrenolm at aia.es>}
+ * @author José Antonio Marqués {@literal <marquesja at aia.es>}
+ */
+public class RegulatedTerminalControllers {
+
+    private final Network network;
+    private final Map<Terminal, List<Identifiable<?>>> controllers;
```

Comment:
With custom IIDM implementations, comparing 2 objects could be problematic.
To avoid problems, `Terminal`s  should be compared on their `Connectable` ids.

In this class, one way to achieve it could be to:
- mapped the controllers by connectable Ids (String);
- adapt the `controller.contains` tests accordingly (in `usedAsRegulatedTerminal` and `replaceRegulatedTerminal`).

Reply:
Done